nncf/model_creation.py

```diff
@@ -141,4 +143,17 @@ def create_compressed_model(model: Module, config: NNCFConfig,
             graph = compressed_graph_builder.build_graph(compressed_model, compressed_model.get_tracing_context())
             graph.visualize_graph(osp.join(config.get("log_dir", "."), "compressed_graph.dot"))
 
+    # Synchronize all processes if run in distributed mode
+    if is_dist_avail_and_initialized():
+        try:
+            barrier()
+        except RuntimeError as err:
```

Comment:
In which cases may a `RuntimeError` occur during a barrier call? Can NNCF really continue working in this scenario?

Reply:
According to a source code- https://github.com/pytorch/pytorch/blob/master/torch/distributed/distributed_c10d.py#L2483
`barrier()` handles `RuntimeError` if a DDP backend doesn't support this function. For example, gloo backend doens't provide a suppor of  `barrier()` for GPU.
If the `RuntimeError` occurs NNCF can not guarantee that the processes are synchronized after `create_compressed_model()`, but in many cases, it should work fine (where working pipeline with a model doesn't demand process synchronization, e.g. examples in NNCF).

Reply:
Please add a short comment in code explaining this situation.